### PR TITLE
Improving test coverage for kubelet/kuberuntime.

### PR DESCRIPTION
**Pull request by enxebre**
**What this PR does / why we need it**:
Increases test coverage for kubelet/kuberuntime 
https://github.com/kubernetes/kubernetes/issues/46123

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/kubernetes/kubernetes/issues/46123

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by yujuhong** [comment]
@feiskyer can you help review the PR?

File: pkg/kubelet/kuberuntime/instrumented_services_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func TestRecordOperation(t *testing.T) {
+
```
Comment (feiskyer) [inline pkg/kubelet/kuberuntime/instrumented_services_test.go]: nit: remove extra empty lines

File: pkg/kubelet/kuberuntime/instrumented_services_test.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	runtimeOperationsCounterExpected := "kubelet_runtime_operations{operation_type=\"create_container\"} 1"
+	runtimeOperationsLatencyExpected := "kubelet_runtime_operations_latency_microseconds_count{operation_type=\"create_container\"} 1"
+
+	assert.HTTPBodyContains(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mux.ServeHTTP(w, r)
+	}), "GET", prometheusUrl, nil, runtimeOperationsCounterExpected)
+
+	assert.HTTPBodyContains(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mux.ServeHTTP(w, r)
+	}), "GET", prometheusUrl, nil, runtimeOperationsLatencyExpected)
```
Comment (feiskyer) [inline pkg/kubelet/kuberuntime/instrumented_services_test.go]: shutdown the http server after all cases.

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go [hunk 3]
```diff
@@ -165,3 +166,94 @@ func TestToKubeContainerStatus(t *testing.T) {
 		assert.Equal(t, test.expected, actual, desc)
 	}
 }
+
+func makeExpetectedConfig(m *kubeGenericRuntimeManager, pod *v1.Pod) *runtimeapi.ContainerConfig {
+
```
Comment (feiskyer) [inline pkg/kubelet/kuberuntime/kuberuntime_container_test.go]: nit: remove extra empty line

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go [hunk 4]
```diff
@@ -165,3 +166,94 @@ func TestToKubeContainerStatus(t *testing.T) {
...
+	container := &pod.Spec.Containers[0]
```
Comment (feiskyer) [inline pkg/kubelet/kuberuntime/kuberuntime_container_test.go]: Also add container index as an arg?

**Comment by feiskyer** [comment]
@k8s-bot ok to test

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go (hunk 4 above)
Comment (enxebre) [inline pkg/kubelet/kuberuntime/kuberuntime_container_test.go]: I like that

**Comment by enxebre** [comment]
thanks for feedback @feiskyer, changes made according suggestions

**Comment by feiskyer** [comment]
should run `./hack/update-bazel.sh` for passing jenkins.

**Comment by enxebre** [comment]
@k8s-bot pull-kubernetes-unit test this

**Comment by enxebre** [comment]
After checking the logs I'm not sure why is failing

File: pkg/kubelet/kuberuntime/helpers_test.go [hunk 5]
```diff
@@ -122,3 +122,86 @@ func TestToKubeContainer(t *testing.T) {
...
+				username: "testUser",
+				err:      nil,
+			},
+		},
+		{
+			"image with uid should return (*int64, \"\", nil)",
+			image{
+				name: "test-image-ref3",
+				uid: &runtimeapi.Int64Value{
+					2,
```
Comment (feiskyer) [inline pkg/kubelet/kuberuntime/helpers_test.go]:
```
&runtimeapi.Int64Value{
  Value: 2,
},
```

**Comment by feiskyer** [comment]
LGTM. Please squash the commits.

**Comment by enxebre** [comment]
commits squashed! thanks for reviewing

**Comment by feiskyer** [comment]
Thanks. Waiting for CI green...

**Pull request merged**